src/Data/Data.ts

```diff
@@ -191,21 +191,15 @@ export async function unsubscribeDataSender(
 
 export function initSocketClient(node: NodeList.ConsensusNodeInfo): void {
   if (config.VERBOSE) Logger.mainLogger.debug('Node Info to socket connect', node)
-  const socketClient = ioclient.connect(`http://${node.ip}:${node.port}`)
+  const socketClient = ioclient.connect(`http://${node.ip}:${node.port}`, {
+    query: {
+      data: JSON.stringify(Crypto.sign({ publicKey: State.getNodeInfo().publicKey })),
+    },
+  })
   socketClients.set(node.publicKey, socketClient)
 
-  let archiverKeyisEmitted = false
-
   socketClient.on('connect', () => {
-    Logger.mainLogger.debug(
-      `${!archiverKeyisEmitted ? 'New connection' : 'Reconnection'} to consensus node ${node.ip}:${
-        node.port
-      } is made`
-    )
-    if (archiverKeyisEmitted) return
-    // Send ehlo event right after connect:
-    socketClient.emit('ARCHIVER_PUBLIC_KEY', config.ARCHIVER_PUBLIC_KEY)
```

Comment:
note there is another spot in archive server where we emit:
 socketClient.emit('ARCHIVER_PUBLIC_KEY', config.ARCHIVER_PUBLIC_KEY)
 src/archivedCycle/StateMetaData.ts
 
 should we remove that also

Reply:
We no longer invoke any code or functions under **`StateMetaData.ts`** module anymore, this is governed by the **`config.experimentalSnapshot`** flag which is set to true perpetually now.
This flag also [disables a set of APIs](https://github.com/shardeum/archive-server/blob/BLUE-257/src/API.ts#L1120-L1202) in the code.